Library/Homebrew/test/cask/system_command_spec.rb

```diff
@@ -1,4 +1,65 @@
 describe Hbc::SystemCommand, :cask do
+  describe "#initialize" do
+    let(:env_args) { ["bash", "-c", 'printf "%s" "${A?}" "${B?}" "${C?}"'] }
+
+    describe "given some environment variables" do
+      subject {
+        described_class.new(
+          "env",
+          args: env_args,
+          env: { "A" => "1", "B" => "2", "C" => "3" },
+          must_succeed: true,
+        )
+      }
+
+      its("run!.stdout") { is_expected.to eq("123") }
+
+      describe "the resulting command line" do
+        it "does not include the given variables" do
+          allow(Open3)
+            .to receive(:popen3)
+            .and_wrap_original do |original_popen3, *args, &block|
+              expect(args).to contain_exactly(
+                a_hash_including("PATH"), ["env"] * 2, *env_args, {}
+              )
+              original_popen3.call("/usr/bin/true", &block)
+            end
+
+          subject.run!
+        end
+      end
+    end
+
+    describe "given some environment variables and sudo: true" do
+      subject {
+        described_class.new(
+          "env",
+          args: env_args,
+          env: { "A" => "1", "B" => "2", "C" => "3" },
+          must_succeed: true,
+          sudo: true,
+        )
+      }
+
+      describe "the resulting command line" do
+        it "includes the given variables explicitly" do
+          allow(Open3)
```

Comment:
I think this could be written more concisely with `expect(…).to receive`.

Reply:
Good point about `expect`; not sure regarding conciseness though.

What would be a more concise way to intercept the `popen3` call?